### PR TITLE
Initialize Rails app

### LINKED ISSUE #5: Create the web app
Copied from https://github.com/GraphQL-Query-Planner/project/issues/1

>We need to create a Rails app we'll be using to test our Query Planner, we should get it up and running on the latest version of Rails (v5.1) and the latest version of Ruby (v2.4.1).

>We'll also want to alway make sure we're using an up to date version of [`graphql-ruby`](https://github.com/rmosolgo/graphql-ruby).

>Since we'll be working with the underlying data layer a lot we'll want to make sure we all have the same version of MySQL installed. The [download page](https://dev.mysql.com/downloads) lists v5.7.19 as the latest and that seems to be the same version homebrew installs.

### PR DESCRIPTION
**Pull request by ismail-syed**
Closes #5 

A recreation of https://github.com/GraphQL-Query-Planner/project/pull/4 with Justin's comments addressed. 

* Added `bundle install` instructions in setup`README.md`
* Added `graphql` gem.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+git_source(:github) do |repo_name|
+  repo_name = "#{repo_name}/#{repo_name}" unless repo_name.include?("/")
+  "https://github.com/#{repo_name}.git"
+end
+
+
+# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
+gem 'rails', '~> 5.1.4'
+# Use sqlite3 as the database for Active Record
+gem 'sqlite3'
```
Comment (DerekStride) [inline Gemfile:12]: We'll probably want `mysql2` but this can be a separate issue.

File: Gemfile [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+gem 'puma', '~> 3.7'
+# Use SCSS for stylesheets
+gem 'sass-rails', '~> 5.0'
+# Use Uglifier as compressor for JavaScript assets
+gem 'uglifier', '>= 1.3.0'
+# See https://github.com/rails/execjs#readme for more supported runtimes
+# gem 'therubyracer', platforms: :ruby
+
+# Use CoffeeScript for .coffee assets and views
+gem 'coffee-rails', '~> 4.2'
```
Comment (DerekStride) [inline Gemfile]: 🔥 ?

File: Gemfile [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+# gem 'bcrypt', '~> 3.1.7'
+
+# Use Capistrano for deployment
+# gem 'capistrano-rails', group: :development
+
+gem 'graphql'
+
+group :development, :test do
+  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
+  gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
```
Comment (DerekStride) [inline Gemfile]: I prefer `pry-byebug` so maybe add that or replace this.

File: Gemfile [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  gem 'selenium-webdriver'
+end
+
+group :development do
+  # Access an IRB console on exception pages or by using <%= console %> anywhere in the code.
+  gem 'web-console', '>= 3.3.0'
+  gem 'listen', '>= 3.0.5', '< 3.2'
+  # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
+  gem 'spring'
+  gem 'spring-watcher-listen', '~> 2.0.0'
```
Comment (DerekStride) [inline Gemfile:53]: Plz behave spring 🙏

File: Gemfile [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+gem 'jbuilder', '~> 2.5'
+# Use Redis adapter to run Action Cable in production
+# gem 'redis', '~> 3.0'
+# Use ActiveModel has_secure_password
+# gem 'bcrypt', '~> 3.1.7'
+
+# Use Capistrano for deployment
+# gem 'capistrano-rails', group: :development
+
+gem 'graphql'
```
Comment (DerekStride) [inline Gemfile]: We'll also need `gem 'graphiql-rails'`

File: Gemfile (hunk 4 above)
Comment (ismail-syed) [reply Gemfile:53]: haha.

File: Gemfile (hunk 1 above)
Comment (ismail-syed) [reply Gemfile:12]: https://github.com/GraphQL-Query-Planner/webscale/issues/4

**Comment by ismail-syed** [comment]
@DerekStride PR fixes addressed.

**Comment by justinkrol** [comment]
From our conversation IRL on Friday, do we want to do `rails new` with the  `--webpack=react` option?

We could probably even do with just `--api` as well if we're using React.

Thoughts?

**Comment by DerekStride** [comment]
I'd be 👍  for `--webpack=react`. I think we can convert it later though.
https://github.com/rails/webpacker#installation

**Review by DerekStride — APPROVED** [review]

**Review by justinkrol — APPROVED** [review]

**Comment by ismail-syed** [comment]
Let's add the `--webpack=react` later down the road when we setup webpack and react.

**Pull request merged**